### ISSUE TITLE
Infinitely loading resource pack.

### ISSUE THREAD

**Issue by ChefJisoo** [PluginBugs/Issues-ItemsAdder#2475]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.

### Discord tag (optional)

Chef Jisoo#7199

### What happened?

Hi! For months now, we've had players with their resource pack (we self host) infinite load upon joining. We have finally found a common link in that everyone affected has played on their account, the pack has worked fine. Then for whatever reason, they'd get a new laptop/pc, try play on that and have the pack reload infinitely.

### Steps to reproduce the issue

No other way I can describe it other than getting a brand new laptop, and trying to download the pack.

### Server version

1.18.2

### ItemsAdder Version

ItemsAdder_3.4.1-r2.jar

### ProtocolLib Version

The /version command does not work, however it is the most recent version added to the website as of 30/03/2023

### LoneLibs Version

The /version command does not work, however it is the most recent version added to the website as of 30/03/2023

### FULL server log

[latest.log](https://github.com/PluginBugs/Issues-ItemsAdder/files/11113062/latest.log)

### Error (optional)

```shell
[16:33:31] [Render thread/INFO]: [CHAT] Events *AstroSheeeep [D+]: Wait, but Alan got the most chests?
[16:33:35] [Render thread/INFO]: [CHAT] AlanPro7571: yeah
[16:33:35] [Worker-Main-26/INFO]: [OptiFine] Mipmap levels: 4
[16:33:36] [Render thread/INFO]: Caught error loading resourcepacks, removing all selected resourcepacks
java.util.concurrent.CompletionException: z: Stitching
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
```

### Screenshots/Videos (you can drag and drop files or paste links)

https://user-images.githubusercontent.com/109172266/228880009-02047530-060b-4b2e-b7a0-913a5080a816.mov

**Comment by LoneDev6** [comment]
> [16:33:31] [Render thread/INFO]: [CHAT] Events *AstroSheeeep [D+]: Wait, but Alan got the most chests?
[16:33:35] [Render thread/INFO]: [CHAT] AlanPro7571: yeah
[16:33:35] [Worker-Main-26/INFO]: [OptiFine] Mipmap levels: 4
[16:33:36] [Render thread/INFO]: Caught error loading resourcepacks, removing all selected resourcepacks
java.util.concurrent.CompletionException: z: Stitching
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]


This error is a bout the fact that you have too many HQ textures which are causing the game to reach the GPU max texture size.
Note: the game merges all your resource pack textures into one, this is what it does behind the scenes.

Reduce your resource pack textures size.
This is not an ItemsAdder issue, it's how the game works.

Without the full CLIENT log that's all I can say.

**Comment by LoneDev6** [comment]
Checking your username now I found our old chat.
I already explained for over a month what the issue is on Discord.
I don't see why you and your team still keep on ignoring all my advices on how to fix the issue (which are not that hard to follow).

**Comment by Tifffy** [comment]
Hello, as a person experiencing this we have also said to you on many, many times it is nothing to do with textures. I am able to load in packs such as and feel free to take a look the Loverfella or MCParks servers that have much more complex packs with much larger textures.

The issue isn't the fact it is getting stuck on one model, it is the fact the only players experiencing the issue are players who would have downloaded the pack on an older device. But their new device says no. Other new players who have never logged on before can log on and download the pack totally fine. I don't know how much clearer this can be. For some reason itemsadder as a pack is saying no to downloading the pack again for users who have downloaded it on other devices in the past.

If it is because you don't know the answer, I would far rather you said that than keep issuing the same response when we all know that isn't it....

**Comment by LoneDev6** [comment]
Please do not comment on closed issues which are not related to your issue.
The error reported by the OP is about what I said:

> This error is a bout the fact that you have too many HQ textures which are causing the game to reach the GPU max texture size.
Note: the game merges all your resource pack textures into one, this is what it does behind the scenes.

This is what the error is telling: `java.util.concurrent.CompletionException: z: Stitching`
